components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1736,6 +1737,146 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             .map((s) => s.url);
     }
 
+    public async getSuggestedRepositories(ctx: TraceContext, organizationId: string): Promise<SuggestedRepository[]> {
```

Comment:
Let's move this method into a new service `SCMService` and call it from here.

Reply:
Looks like we have an [ScmService](https://github.com/gitpod-io/gitpod/blob/bef25a6a722178af01939c6b3c3ab0fa8d6c6bb3/components/server/src/projects/scm-service.ts#L15) so I'll go ahead and add it there. Let me know if that doesn't seem like the right place.

Reply:
What would you suggest for the few methods from gitpod-server-impl that this currently depends on?

* [getAuthProviders()](https://github.com/gitpod-io/gitpod/blob/bef25a6a722178af01939c6b3c3ab0fa8d6c6bb3/components/server/src/workspace/gitpod-server-impl.ts#L753) - Not sure on this one, should it all move to the AuthProviderService behind a method there?
* [getWorkspaces()](https://github.com/gitpod-io/gitpod/blob/bef25a6a722178af01939c6b3c3ab0fa8d6c6bb3/components/server/src/workspace/gitpod-server-impl.ts#L1071) - mostly just a call to `workspaceService.getWorkspaces()` except for the guardAccess checks against each workspace.

Reply:
I went down the route of pulling it into the SCMService and getAuthProviders into AuthProviderService, but then bumped into cyclic deps issues between SCMSerivce/AuthProviderService and ProjectService, so backed up and will consider that a future improvement.